klayout/drc/rule_decks/main.drc

```diff
@@ -14,880 +14,763 @@
-
-      metal4_slot   = get_polygons(46 , 3 )
-      count   = metal4_slot.count()
-      logger.info("metal4_slot has %d polygons" % [count])
-      polygons_count  += count
-
-      metal4_blk    = get_polygons(46 , 5 )
-      count   = metal4_blk.count()
-      logger.info("metal4_blk has %d polygons" % [count])
-      polygons_count  += count
-
-      via4          = get_polygons(41 , 0 )
-      count   = via4.count()
-      logger.info("via4 has %d polygons" % [count])
-      polygons_count  += count
-
-      if METAL_LEVEL == "5LM"
-        metal5_drawn  = get_polygons(81 , 0 )
-        count   = metal5_drawn.count()
-        logger.info("metal5_drawn has %d polygons" % [count])
-        polygons_count  += count
-
-        metal5_dummy  = get_polygons(81 , 4 )
-        count   = metal5_dummy.count()
-        logger.info("metal5_dummy has %d polygons" % [count])
-        polygons_count  += count
-
-        metal5        = metal5_drawn + metal5_dummy
-
-        metal5_label  = get_polygons(81 , 10)
-        count   = metal5_label.count()
-        logger.info("metal5_label has %d polygons" % [count])
-        polygons_count  += count
-      
-        metal5_slot   = get_polygons(81 , 3 )
-        count   = metal5_slot.count()
-        logger.info("metal5_slot has %d polygons" % [count])
-        polygons_count  += count
-
-        metal5_blk    = get_polygons(81 , 5 )
-        count   = metal5_blk.count()
-        logger.info("metal5_blk has %d polygons" % [count])
-        polygons_count  += count
+
+      via4 = get_polygons(41, 0)
+      count = via4.count
+      logger.info("via4 has #{count} polygons")
+      polygons_count += count
+
+      case METAL_LEVEL
+      when '5LM'
+        metal5_drawn = get_polygons(81, 0)
+        count = metal5_drawn.count
+        logger.info("metal5_drawn has #{count} polygons")
+        polygons_count += count
+
+        metal5_dummy = get_polygons(81, 4)
+        count = metal5_dummy.count
+        logger.info("metal5_dummy has #{count} polygons")
+        polygons_count += count
+
+        metal5 = metal5_drawn + metal5_dummy
+
+        metal5_label = get_polygons(81, 10)
+        count = metal5_label.count
+        logger.info("metal5_label has #{count} polygons")
+        polygons_count += count
+
+        metal5_slot = get_polygons(81, 3)
+        count = metal5_slot.count
+        logger.info("metal5_slot has #{count} polygons")
+        polygons_count += count
+
+        metal5_blk = get_polygons(81, 5)
+        count = metal5_blk.count
+        logger.info("metal5_blk has #{count} polygons")
+        polygons_count += count
 
         top_via       = via4
         topmin1_via   = via3
         top_metal     = metal5
         topmin1_metal = metal4
-      elsif METAL_LEVEL == "6LM"
-        metal5_drawn   = get_polygons(81 , 0 )
-        count   = metal5_drawn.count()
-        logger.info("metal5_drawn has %d polygons" % [count])
-        polygons_count  += count
+      when '6LM'
```

Comment:
seems weird to switch from if/else to a case/when in the middle of block?

Reply:
Same as above

Reply:
@proppy Referring to #42